js/common/lib/backend.ts

```diff
@@ -14,23 +15,49 @@ export declare namespace SessionHandler {
 }
 
 /**
- * Represent a handler instance of an inference session.
+ * Represents shared SessionHandler functionality
  *
  * @ignore
  */
-export interface SessionHandler {
+interface SessionHandler {
   dispose(): Promise<void>;
 
   readonly inputNames: readonly string[];
   readonly outputNames: readonly string[];
+}
 
+/**
+ * Represent a handler instance of an inference session.
+ *
+ * @internal
+ */
+export interface InferenceSessionHandler extends SessionHandler {
   startProfiling(): void;
   endProfiling(): void;
 
   run(feeds: SessionHandler.FeedsType, fetches: SessionHandler.FetchesType,
       options: InferenceSession.RunOptions): Promise<SessionHandler.ReturnType>;
 }
 
+/**
+ * Represent a handler instance of a training inference session.
+ *
+ * @internal
+ */
+export interface TrainingSessionHandler extends SessionHandler {
+  dispose(): Promise<void>;
+
+  readonly inputNames: readonly string[];
+  readonly outputNames: readonly string[];
+
+  runTrainStep(
+      feeds: SessionHandler.FeedsType, fetches: SessionHandler.FetchesType,
+      options: InferenceSession.RunOptions): Promise<SessionHandler.ReturnType>;
+
+  loadParametersBuffer(buffer: ArrayBufferLike, trainableOnly: boolean): Promise<void>;
```

Comment:
ArrayBufferLike does not work for this. You need to use it with a memory view instead of the whole memory buffer. Use `buffer: Uint8Array` or `arrayBuffer: ArrayBufferLike, byteOffset: number, byteLength: number` instead.

Reply:
same to `getContiguousParameters`